jnosql-dynamodb/src/main/java/org/eclipse/jnosql/databases/dynamodb/communication/DefaultDynamoDBDocumentManager.java

```diff
@@ -0,0 +1,348 @@
+                .map(this::update)
+                .toList();
+    }
+
+    @Override
+    public void delete(DocumentDeleteQuery documentDeleteQuery) {
+        Objects.requireNonNull(documentDeleteQuery, "documentDeleteQuery is required");
+
+        List<String> primaryKeys = getDescribeTableResponse(documentDeleteQuery.name())
+                .table()
+                .keySchema()
+                .stream()
+                .map(KeySchemaElement::attributeName).toList();
+
+        DocumentQuery.DocumentQueryBuilder selectQueryBuilder = DocumentQuery.builder()
+                .select(primaryKeys.toArray(new String[0]))
+                .from(documentDeleteQuery.name());
+
+        documentDeleteQuery.condition().ifPresent(selectQueryBuilder::where);
+
+        select(selectQueryBuilder.build()).forEach(
+                documentEntity ->
+                        dynamoDbClient().deleteItem(DeleteItemRequest.builder()
+                                .tableName(documentDeleteQuery.name())
+                                .key(getItemKey(documentEntity))
+                                .build()));
+    }
+
+    @Override
+    public Stream<DocumentEntity> select(DocumentQuery documentQuery) {
+        Objects.requireNonNull(documentQuery, "documentQuery is required");
+        DynamoDBQuery dynamoDBQuery = DynamoDBQuery
+                .builderOf(documentQuery.name(), getEntityAttributeName(), documentQuery)
+                .get();
+
+        ScanRequest.Builder selectRequest = ScanRequest.builder()
+                .consistentRead(true)
+                .tableName(dynamoDBQuery.table())
+                .projectionExpression(dynamoDBQuery.projectionExpression())
+                .filterExpression(dynamoDBQuery.filterExpression())
+                .expressionAttributeNames(dynamoDBQuery.expressionAttributeNames())
+                .expressionAttributeValues(dynamoDBQuery.expressionAttributeValues())
+                .select(dynamoDBQuery.projectionExpression() != null ? Select.SPECIFIC_ATTRIBUTES : Select.ALL_ATTRIBUTES);
+
+        return StreamSupport
+                .stream(dynamoDbClient().scanPaginator(selectRequest.build()).spliterator(), false)
+                .flatMap(scanResponse -> scanResponse.items().stream()
+                        .map(item -> toDocumentEntity(this::resolveEntityNameAttributeName, item)));
+    }
+
+    @Override
+    public long count(String tableName) {
+        Objects.requireNonNull(tableName, "tableName is required");
+        try {
+            return getDescribeTableResponse(tableName)
+                    .table()
+                    .itemCount();
+        } catch (ResourceNotFoundException ex) {
+            return 0;
+        }
+    }
+
+    @Override
+    public void close() {
+        this.dynamoDbClient.close();
+    }
+
+    @Override
+    public Stream<DocumentEntity> partiQL(String query) {
+        return partiQL(query,new Object[0]);
```

Comment:
Once it is varags, why not leave it empty?

```suggestion
        return partiQL(query);
```

Reply:
Yeah! I fixed it!